### ISSUE TITLE
Error when drawing newTextBatch object in Love 12.0

### ISSUE THREAD

**Issue by modcos** [love2d/love#1922]
This example reproduces the error:
Error: main.lua:16: Texture's type (unknown) must match the type of the shader's main texture type (2d).
This happens randomly at startup. If I remove love.graphics.print, then no error occurs.
Highdpi must be enabled.

```lua
local font = love.graphics.getFont()
local text_batch = love.graphics.newText(font)

local text = {
	"If you get stuck, many friendly people are ready",
	"to help you at the forums. Be warned, however,",
	"that it sometimes gets too friendly."
}

for _, v in ipairs(text) do
	text_batch:add(v)
end

function love.draw()
	love.graphics.print('ABCDEFG')
	love.graphics.draw(text_batch)
end
```

**Comment by MikuAuahDark** [comment]
Related issue: #1914.

**Comment by slime73** [comment]
I ran the example (plus a conf.lua with highdpi enabled) 20 times on my Mac with a high DPI display and didn't get any error. Can you provide more details about the system you're running on? What's the typical rate of errors compared to successful startups for you?

**Comment by modcos** [comment]
I recorded a video demonstration of the error, it occurs in 90% of cases. Tested on Windows 10, 4090 graphics card.
https://drive.google.com/file/d/10hnCRQvXjpDq-6r2xm15ppmIbS7Nav_a/view?usp=share_link

The error seems to depend directly on the desktop resolution. When I use 3840x2160 I get an error. But if I set the video mode to 1920x1080 the error does not occur. 
I also tried to get the error on a Windows 10 tablet with an intel hd card at 2160x1440 resolution, but it didn't work. So this led me to believe that it is the high resolution of the screen that is causing the error.

**Comment by slime73** [comment]
Thanks. I don't have a 4k monitor but I can probably try a few things to hopefully hit the same issue that happens there.

Do you know what the DPI scale is set to in the Windows Display Settings when you use a 4k resolution? (I'd guess 200% but maybe it's different.)

**Comment by modcos** [comment]
I use 250%.

**Comment by slime73** [comment]
Thanks! I managed to reproduce it, and I committed a fix in [90b22f7](https://github.com/love2d/love/commit/90b22f7d42f953dbf4a2bc720b4a6e7d4b8c7b5a).
